### PR TITLE
Migrate `NewMedicalHistoryScreen` to a Fragment

### PR DESCRIPTION
**Pull request by Honey14**
https://app.clubhouse.io/simpledotorg/story/4532/migrate-newmedicalhistoryscreen-to-a-fragment

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt [hunk 1]
```diff
@@ -172,6 +174,24 @@ class NewMedicalHistoryScreen : BaseScreen<
     }
   }
 
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    requireContext().injector<Injector>().inject(this)
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt:160]: Just an FYI, we can use the `context` from the `onAttach` directly. There is no need to use `requireContext` here. It doesn't make any difference, just a visual consistency with other screens in the project.

File: app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt [hunk 2]
```diff
@@ -164,8 +168,8 @@ class NewMedicalHistoryScreen : BaseScreen<
       router.pop()
     }
 
-    post {
-      hideKeyboard()
+    rootView.post {
+      rootView.hideKeyboard()
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt:172]: Previously we were doing this to hide the keyboard when coming from the patient entry screen. Can we verify if it's still needed? or else we can remove this code as well.

File: app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt (hunk 1 above)
Comment (Honey14) [reply app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt:160]: Oh right, my bad.

**Review by janhavisinghh — APPROVED** [review]
LGTM

**Pull request merged**